dev/main.c

```diff
@@ -0,0 +1,146 @@
+#include "main.h"
+#include "Players/players.h"
+#include "Tiles/entity.h"
+//#include "Stages/stage1.h"
+
+unsigned char frame_counter;
+unsigned char volume_atenuation;
+
+Entity* entity;
+
+void loadGraphics2vram(void)
+{
+  /* clear VRAM */
+  SMS_VRAMmemsetW(0, 0x0000, 0x4000);
+
+  // backgound
+  SMS_loadBGPalette(background_palette_bin);
+
+  SMS_loadPSGaidencompressedTiles(backgroundtiles_psgcompr, 0);
+  SMS_loadTileMap(0,0, backgroundtilemap_bin, backgroundtilemap_bin_size);
+  //SMS_setBGPaletteColor(0, RGB(0, 2, 3));
+
+  // items
+  SMS_loadPSGaidencompressedTiles(items_tiles_psgcompr, 18);
+
+  // explosions
+  // SMS_loadPSGaidencompressedTiles(explosions_tiles_psgcompr, 41);
+
+
+  // sprites
+  SMS_loadSpritePalette(spritepalette_bin);
+  SMS_loadTiles(spritetiles_down_bin, PLAYER1_SPRITE_POSITION, 32*6*6);
+  //SMS_setSpritePaletteColor(0, RGB(0, 0, 0));
+
+  // tilelist
+  Tile tilelist1[] = 
+  {
+    // frame 1
+    {11, 0, 0, 0, 0, 1},
+    {10, 0, 0, 0, 0, 1},
+    {11, 0, 0, 0, 0, 1},
+    {10, 0, 0, 0, 0, 1},
+  };
+
+  Tile tilelist2[] = 
+  {
+    // frame 2
+    {20, 0, 0, 0, 0, 1},
+    {21, 0, 0, 0, 0, 1},
+    {20, 0, 0, 0, 0, 1},
+    {21, 0, 0, 0, 0, 1}
+  };
+  
+  MetaTile* metatile1 = MetaTile_Create(tilelist1, 4, 2, 2);
+  MetaTile* metatile2 = MetaTile_Create(tilelist2, 4, 2, 2);
+
+  MetaTile* metatilelist[] = 
+  {
+    metatile1,
+    metatile2
+  };
+
+  entity = Entity_Create(metatilelist, 2);
```

Comment:
Now this does look like a problem with dangling pointers to me.

Both `metatile1` and `metatile2` as well as `metatilelist` are local variables to `loadGraphics2vram` on the stack, and the pointers to them will become invalid once it returns, yet inside the implementation of `Entity_Create` you are copying the pointers.

Without following the logic through, I can't tell if that actually matters, but the way it's written doesn't lead me to believe you expect those pointers to be invalid.

Reply:
It's very possible and even probable. This was one area where I didn't feel confident about how this aught to work. I don't know the general way to handle these types of structures and was definitely an area why I reached out on the forum.